### PR TITLE
workflow test for revit node

### PR DESCRIPTION
**Pull request by zora-wang**
workflow
@riteshchandawar ,@monika.

### CODE REVIEW

File: test/Libraries/RevitIntegrationTests/WorkflowTests.cs [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+            }
+
+        }
+
+
+        [Test]
+        [TestModel(@".\Workflow\DynamoRevit\DynamoSample.rvt")]
+        public void Test_AdaptiveComponent1()
+        {
+            string samplePath = Path.Combine(workingDirectory, @".\Workflow\DynamoRevit\02_Adaptive Component Placement 1.dyn");
```
Comment (riteshchandawar) [inline test/Libraries/RevitIntegrationTests/WorkflowTests.cs]: Always make these kind of long statements in two lines so that, it can be readable on PR.

Comment (zora-wang) [inline test/Libraries/RevitIntegrationTests/WorkflowTests.cs]: okay

File: test/Libraries/RevitIntegrationTests/WorkflowTests.cs [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+            Assert.AreEqual(11, model.CurrentWorkspace.Nodes.Count);
+            Assert.AreEqual(10, model.CurrentWorkspace.Connectors.Count());
+
+            //check AdaptiveComponent.ByPoint
+            var AC = "357e7a53-361c-4c1e-81ae-83e16213a39a";
+            AssertPreviewCount(AC, 9);
+
+            //check elements in AdaptiveComponent.ByPoint are not null
+            for (int i = 0; i < 9; i++)
+            {
```
Comment (riteshchandawar) [inline test/Libraries/RevitIntegrationTests/WorkflowTests.cs]: @line 125 preview count is 9 and in for loop range is from 0 to 9 which is 10 element. How come this test case is passing?

File: test/Libraries/RevitIntegrationTests/WorkflowTests.cs [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+            Assert.AreEqual(17, model.CurrentWorkspace.Nodes.Count);
+            Assert.AreEqual(19, model.CurrentWorkspace.Connectors.Count());
+
+            //check AdaptiveComponent.ByPoint
+            var AC = "357e7a53-361c-4c1e-81ae-83e16213a39a";
+            AssertPreviewCount(AC, 9);
+
+            //check elements in AdaptiveComponent.ByPoint are not null
+            for (int i = 0; i < 9; i++)
+            {
```
Comment (riteshchandawar) [inline test/Libraries/RevitIntegrationTests/WorkflowTests.cs]: At line 149 preview count is 9 and in for loop range is from 0 to 9 which is 10 element. How come this test case is passing?

Comment (zora-wang) [inline test/Libraries/RevitIntegrationTests/WorkflowTests.cs]: i<9 ; not i<=9 ; os range is from 0 to 8

File: test/Libraries/RevitIntegrationTests/WorkflowTests.cs [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+            string samplePath = Path.Combine(workingDirectory, @".\Workflow\DynamoRevit\04_ImportSolid.dyn");
+            string testPath = Path.GetFullPath(samplePath);
+            ViewModel.OpenCommand.Execute(testPath);
+            RunCurrentModel();
+            var model = ViewModel.Model;
+            Assert.AreEqual(16, model.CurrentWorkspace.Nodes.Count);
+            Assert.AreEqual(16, model.CurrentWorkspace.Connectors.Count());
+
+            //check elements in ImportInstance.ByGeometries are not null
+            Assert.IsNotNull(GetPreviewValue("e3fedc00-247a-4971-901c-7fcb063344c6"));
```
Comment (riteshchandawar) [inline test/Libraries/RevitIntegrationTests/WorkflowTests.cs]: it should be validating what should be the type of output. Please add "as Solid" or whatever output your are getting from ImportGeometries node.

File: test/System/Workflow/DynamoRevit/03_Adaptive Component Placement 2.dyn [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    <Dynamo.Models.ConnectorModel start="3e193e54-161c-41ea-b3fa-1af0d781545b" start_index="0" end="f4ebe6f9-b7d6-4c43-989f-8008f6f949b3" end_index="1" portType="0" />
+    <Dynamo.Models.ConnectorModel start="cb642e29-8e32-428a-beb3-10549663a490" start_index="0" end="357e7a53-361c-4c1e-81ae-83e16213a39a" end_index="0" portType="0" />
+  </Connectors>
+  <Notes>
+    <Dynamo.Models.NoteModel text="Select curve from Revit" x="846.35713721073" y="635.617946335688" />
+    <Dynamo.Models.NoteModel text="Select curve from Revit" x="846.35713721073" y="779.687312996161" />
+    <Dynamo.Models.NoteModel text="Select curve from Revit" x="846.35713721073" y="921.415745553684" />
+    <Dynamo.Models.NoteModel text="Run this graph the .RVT file in the Samples folder." x="837.916415601284" y="540.706854239215" />
+  </Notes>
+  <SessionTraceData>
```
Comment (riteshchandawar) [inline test/System/Workflow/DynamoRevit/03_Adaptive Component Placement 2.dyn]: Trace data shouldn't be there.

File: test/System/Workflow/DynamoRevit/02_Adaptive Component Placement 1.dyn [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    <Dynamo.Models.ConnectorModel start="1c420759-4d5f-45b4-8310-f1fa07e92c99" start_index="0" end="83f06d4b-59ed-4888-a259-43d12f3d1736" end_index="0" portType="0" />
+  </Connectors>
+  <Notes>
+    <Dynamo.Models.NoteModel guid="f5e682e4-c6dc-4cad-b589-d68a8d462143" text="Select curve from Revit" x="846.35713721073" y="635.617946335688" />
+    <Dynamo.Models.NoteModel guid="2dd763e1-3140-42bc-ad54-83172116abda" text="Select curve from Revit" x="846.35713721073" y="779.687312996161" />
+    <Dynamo.Models.NoteModel guid="693ca362-4259-4e91-870d-56b7b7652e80" text="Select curve from Revit" x="846.35713721073" y="921.415745553684" />
+    <Dynamo.Models.NoteModel guid="52b0f4d5-d259-4c98-b93f-d41cae027820" text="Run this graph the .RVT file in the Samples folder." x="837.916415601284" y="540.706854239215" />
+  </Notes>
+  <Annotations />
+  <SessionTraceData>
```
Comment (riteshchandawar) [inline test/System/Workflow/DynamoRevit/02_Adaptive Component Placement 1.dyn]: Trace data shouldn't be there.

File: test/Libraries/RevitIntegrationTests/WorkflowTests.cs [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+            string testPath = Path.GetFullPath(samplePath);
+            ViewModel.OpenCommand.Execute(testPath);
+            RunCurrentModel();
+            var model = ViewModel.Model;
+            Assert.AreEqual(39, model.CurrentWorkspace.Nodes.Count);
+            Assert.AreEqual(47, model.CurrentWorkspace.Connectors.Count());
+
+            //check Floor.ByOutLineTypeAndLevel
+            var floor = "1bcce36c-7ea3-4c70-9271-544fd378ec41";
+            AssertPreviewCount(floor, 14);
```
Comment (riteshchandawar) [inline test/Libraries/RevitIntegrationTests/WorkflowTests.cs]: Add for loop to check all Floors from the list.

File: test/Libraries/RevitIntegrationTests/WorkflowTests.cs (hunk 3 above)
Comment (riteshchandawar) [inline test/Libraries/RevitIntegrationTests/WorkflowTests.cs]: ooh ok. Good.

File: test/Libraries/RevitIntegrationTests/WorkflowTests.cs (hunk 2 above)
Comment (riteshchandawar) [inline test/Libraries/RevitIntegrationTests/WorkflowTests.cs]: This should be ok as its <9. Ignore my previous comment.

**Comment by zora-wang** [comment]
@riteshchandawar , the rainbow.jpg added is used in tower test.

**Comment by zora-wang** [comment]
test panel
@riteshchandawar

**Comment by monikaprabhu** [comment]
@zora-wang - Please add detail description for each test with which nodes are being tested.
Also how much time does it take to run all tests?

**Comment by zora-wang** [comment]
pass tests
@riteshchandawar , @monikaprabhu  please help to review it

**Pull request merged**